### DISCUSSION TITLE
Using the IPFS with filecoin

### DISCUSSION THREAD

**Discussion by dale-cfx** [filecoin-project/community#685] [Developers]
I know filecoin is a storage chain with Zk-SNARK protocol. Can you help me understanding of creating an where i use filecoin as the storage option. What all should i understand? How to store? what will be the network fee? how to or where can i access stored files from etc?
please point towards right documentation and details. I have been searching for that for sometime.
Thanks

**Comment by web3jenks** [comment]
Hey @dale-cfx, Developer Advocate Jenks here. This is a common question to Filecoin network. The foundamental layer is designed to for enterprise level of storage arrays and primitives for people to build storage applications for any market or use cases. So depending on the size and use cases of your storage, one will have to find the right tools. The same goes for access stored files (retrieval). The tools used to retrieve 10PB files (physical transfer) are different from 1MB (can be done via decentralise CDN like Saturn). 

Perhaps, we can start with what you would like to store first? Are these small files or very large files (>10TB). 

Then we can recommend some right tools. 

This is a site that can guide you to the right tools, but this does not cover all: 
https://dataonboarding.filecoin.io/

**Reply by dale-cfx** [reply]
I will explain my usecase here.
I'm trying to build a file sharing app. Maximum expected size per document will not be more than 500 MB per upload. This is a B2C solution, so the user base can be high. So i wanted to know how filecoin can help me with this. I hope this also is an EVM based chain for storage. Can i create seperate address for each user where i can utilize it as an account and the user can upload files and share the transaction or CID for retrieving the file? Also how do i retrieve it via program or api?
